src/container/srv_container.c

```diff
@@ -1382,19 +1449,38 @@ cont_open(struct rdb_tx *tx, struct ds_pool_hdl *pool_hdl, struct cont *cont,
 	}
 
 	/* Determine pool meets container redundancy factor requirements */
-	if (!(in->coi_flags & DAOS_COO_FORCE)) {
+	is_healthy = cont_status_is_healthy(prop, &stat_pm_ver);
+	if (is_healthy) {
+		int	rf;
+
+		rf = daos_cont_prop2redunfac(prop);
 		pmap = pool_hdl->sph_pool->sp_map;
-		rc = cont_verify_redun_req(pmap, prop);
-		if (rc != 0) {
-			D_ERROR(DF_CONT": Container does not meet redundancy "
-					"requirements, set DAOS_COO_FORCE to "
-					"force container open rc: %d.\n",
-				DP_CONT(cont->c_svc->cs_pool_uuid,
-					cont->c_uuid), rc);
-			daos_prop_free(prop);
-			D_GOTO(out, rc);
+		rc = pool_map_rf_verify(pmap, stat_pm_ver, rf);
+		if (rc == -DER_RF) {
+			rc1 = cont_status_set_unclean(tx, pool_hdl->sph_pool,
```

Comment:
I thought this should be done whenever the pool map are changing. Otherwise how to check RF for inflight I/O.

I thought the open container will turns to readonly, if the failure targets reach to RF, otherwise we might meet trouble for reintegration?

Reply:
"I thought this should be done whenever the pool map are changing"
we had discuss that before. If iterates and checks all container for each pool map change, it is more expensive , note there possibly with lots of inactive and even will never open conts.

"Otherwise how to check RF for inflight I/O."
sorry I don't understand what's the problem?

"I thought the open container will turns to readonly, if the failure targets reach to RF, otherwise we might meet trouble for reintegration?"
not sure if I understand the problem. but user can force open the cont, or clear the UNCLEAN flag to open it.

Reply:
"Otherwise how to check RF for inflight I/O."
sorry I don't understand what's the problem?

considering this sequence.

1. open container.  RF=1 
2. failure 2 targets,  may lost data, but does not set unclean.
3. continue I/O
4. reintegrate 1 target.  since we do not set unclean data, then it will delete all previous data. 
5. close the container.
6. open the container again. since only 1 fail target, so no unclean property. but you already lose data.

Reply:
OK, we discussed that REINT case before and with you on the meeting, no?
If I understood it correctly, the device can be REINT means that all previous data are accessible, so we don't worry for REINT device for now.

Reply:
"since we do not set unclean data, then it will delete all previous data."
TBH, I am not quite clear for that case. But any way, we can resolve things step by step.

Reply:
Hmm, according to discussion, if I remember correctly, whether the previous data should be deleted depend on the container unclean property.

if the container is clean, then it means no data loss, so it will delete the previous data
during reintegration.

if the container is unclean, then it means data loss, so it will assume the previous data still exists during reintegration.

But in your implementation,  this unclean property can be missed if the container is already opened?

Reply:
What if the administrator do a "prop query" before REINT it? in that case the container will be marked as UNCLEAN.